base/array.jl

```diff
@@ -1216,6 +1218,143 @@ function insert!(a::Array{T,1}, i::Integer, item) where T
     return a
 end
 
+"""
+    inserteach!(a::Vector, index::Integer, items::Vector)
+
+Insert all of the elements of `items` into `a` at the given `index`. `index` is the index of first element of `items` in the resulting `a`.
+
+# Examples
+```jldoctest
+julia> inserteach!([1, 2, 3], 2, [8, 9])
+5-element Array{Int64,1}:
+ 1
+ 8
+ 9
+ 2
+ 3
+```
+"""
+function inserteach!(a::Vector, i::Integer, items::AbstractVector)
```

Comment:
This method does not seem consistent with the others --- in the other cases, the indices and items have the same length.

Reply:
Here the index is a single `Integer`. I renamed `i` to `index` for clarification.

Reply:
The problem isn't the name of the argument, it's the behavior. It's this difference:
```
julia> a[1:1] = [1]
1-element Array{Int64,1}:
 1

julia> a[1] = [1]
ERROR: MethodError: Cannot `convert` an object of type Array{Int64,1} to an object of type Int64
```

Reply:
These are two different methods, they are not meant to be the same:
- `function inserteach!(a::Vector, index::Integer, items::AbstractVector)`:
Inserts all the items at index `i`. 1st item becomes `i`th.

- `function inserteach!(a::Vector, indices::Vector, items::AbstractVector)`:
Inserts the items at the indices given. Indices now has a 1-1 association with items.